### PR TITLE
k8s 1.12 issue workaround

### LINKED ISSUE #3681: kube-scheduler is never created using k8s 1.12 and vmss
**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE

**What version of acs-engine?**:
master commit 1ade2f3

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
k8s 1.12

**What happened**:
Created a new cluster with acs-engine master 1ade2f3 using k8s 1.12 and VMSS resulting in a cluster that does not have kube-scheduler. `docker ps -a` does not show kube-scheduler container even being scheduled. As a result, addon pods are shown as `pending`. New pods scheduled are shown as `pending` without any events in `kubectl describe pod` 

From the master kubelet log, we never see this expected event in the log:
```
SyncLoop (PLEG): "kube-scheduler-k8s-master-36079709-0_kube-system(4112ac0d067b2419602ce8457d86bbe4)", event: &pleg.PodLifecycleEvent{ID:"4112ac0d067b2419602ce8457d86bbe4", Type:"ContainerStarted", Data:"c8f500822117e406ef587d816de0497bce0915d05c29ad38ddb3331f12a8544c"}
```
Restarting the kubelet service on master solves this issue.

**What you expected to happen**:
A healthy cluster is created with k8s 1.12 and VMSS.

**How to reproduce it** (as minimally and precisely as possible):
With the apimodel below, a cluster will be created using VMSS and k8s 1.12.

```
{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.12"
    },
    "masterProfile": {
      "count": 1,
      "dnsPrefix": "",
      "vmSize": "Standard_DS2_v2"
    },
    "agentPoolProfiles": [
      {
        "name": "agentpool",
        "count": 2,
        "vmSize": "Standard_DS2_v2"
      }
    ],
    "linuxProfile": {
      "adminUsername": "azureuser",
      "ssh": {
        "publicKeys": [
          {
            "keyData": ""
          }
        ]
      }
    },
    "servicePrincipalProfile": {
    }
  }
}
```

**Anything else we need to know**:
This issue is only visible in VMSS and k8s 1.12. VMAS works as intended.

cc @feiskyer

### PR DESCRIPTION
**Pull request by ritazh**
Fixes #3681

### CODE REVIEW

File: parts/k8s/kubernetescustomscript.sh [hunk 1]
```diff
@@ -108,6 +108,8 @@ if [[ ! -z "${MASTER_NODE}" ]]; then
     writeKubeConfig
     ensureEtcd
     ensureK8sControlPlane
+    # workaround for 1.12 bug https://github.com/Azure/acs-engine/issues/3681 will remove once upstream is fixed
+    ensureKubelet
```
Comment (jackfrancis) [inline parts/k8s/kubernetescustomscript.sh]: So starting the systemd kubelet service two times in succession on masters fixes this?

Comment (ritazh) [reply parts/k8s/kubernetescustomscript.sh]: yup

**Comment by jackfrancis** [comment]
@CecileRobertMichon does our new CSE implementation allow us to easily do this workaround only on k8s 1.12 versions?

**Comment by CecileRobertMichon** [comment]
> @CecileRobertMichon does our new CSE implementation allow us to easily do this workaround only on k8s 1.12 versions?
--

Yes:

```
# workaround for 1.12 bug https://github.com/Azure/acs-engine/issues/3681 will remove once upstream is fixed 
if [[ "${KUBERNETES_VERSION}" = 1.12.* ]]; then
    ensureKubelet 
fi
```

should work

**Pull request merged**